support-lib/jni/djinni_support.hpp

```diff
@@ -47,6 +47,48 @@ void jniShutdown();
  */
 JNIEnv * jniGetThreadEnv();
 
+JavaVM * getCachedJVM();
+/*
+ * Runs the given function on a Java thread and subsequently detaches it if it was not
+ * previously attached.
+ */
+template<typename Func>
+typename std::result_of<Func()>::type runOnJvmThreadWithReturn(Func f) {
+    auto g_cachedJVM = getCachedJVM();
+    assert(g_cachedJVM);
+    bool attached = false;
+    JNIEnv * env = nullptr;
+    jint get_res = g_cachedJVM->GetEnv(reinterpret_cast<void**>(&env), JNI_VERSION_1_6);
+    if (get_res == JNI_EDETACHED) {
+        get_res = g_cachedJVM->AttachCurrentThread(&env, nullptr);
+        attached = true;
+    }
+    if (get_res != 0 || !env) {
+        std::abort();
+    }
+    auto result = f();
+    if (attached) g_cachedJVM->DetachCurrentThread();
```

Comment:
Please make the if one line if's consistent with the rest, just have a {} for if's scope like the majority.
```diff
-    if (attached) g_cachedJVM->DetachCurrentThread();
+    if (attached) {
+        g_cachedJVM->DetachCurrentThread();
+    }
```